### PR TITLE
New resource: aws_ec2_transit_gateway_vpc_attachment_accepter

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Fixes https://github.com/terraform-providers/terraform-provider-aws/issues/8125.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
FEATURES:

* New Resource: aws_ec2_transit_gateway_vpc_attachment_accepter (#8125)
```

Acceptance tests:

```console
$ AWS_ALTERNATE_ACCESS_KEY_ID=AAAAAAAAAAAAAAAAAAAA AWS_ALTERNATE_SECRET_ACCESS_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx make testacc TEST=./aws/ TESTARGS='-run=estAccAWSEc2TransitGatewayVpcAttachmentAccepter_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=estAccAWSEc2TransitGatewayVpcAttachmentAccepter_ -timeout 120m
=== RUN   TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== PAUSE TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== CONT  TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
--- PASS: TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic (400.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	400.206s
```

### CODE REVIEW

File: aws/resource_aws_ec2_transit_gateway_vpc_attachment.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func resourceAwsEc2TransitGatewayVpcAttachmentCreate(d *schema.ResourceData, met
 
 	d.SetId(aws.StringValue(output.TransitGatewayVpcAttachment.TransitGatewayAttachmentId))
 
-	if err := waitForEc2TransitGatewayRouteTableAttachmentCreation(conn, d.Id()); err != nil {
```
Comment (ewbankkit) [inline aws/resource_aws_ec2_transit_gateway_vpc_attachment.go]: Renamed to reflect true functionality.

File: aws/resource_aws_ec2_transit_gateway_vpc_attachment.go [hunk 2]
```diff
@@ -238,7 +238,7 @@ func resourceAwsEc2TransitGatewayVpcAttachmentUpdate(d *schema.ResourceData, met
 			return fmt.Errorf("error modifying EC2 Transit Gateway VPC Attachment (%s): %s", d.Id(), err)
 		}
 
-		if err := waitForEc2TransitGatewayRouteTableAttachmentUpdate(conn, d.Id()); err != nil {
```
Comment (ewbankkit) [inline aws/resource_aws_ec2_transit_gateway_vpc_attachment.go]: Renamed to reflect true functionality.

File: aws/resource_aws_ec2_transit_gateway_vpc_attachment.go [hunk 3]
```diff
@@ -295,7 +295,7 @@ func resourceAwsEc2TransitGatewayVpcAttachmentDelete(d *schema.ResourceData, met
 		return fmt.Errorf("error deleting EC2 Transit Gateway VPC Attachment: %s", err)
 	}
 
-	if err := waitForEc2TransitGatewayRouteTableAttachmentDeletion(conn, d.Id()); err != nil {
```
Comment (ewbankkit) [inline aws/resource_aws_ec2_transit_gateway_vpc_attachment.go]: Renamed to reflect true functionality.

File: website/docs/r/ec2_transit_gateway_vpc_attachment_accepter.html.markdown [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+Manages the accepter's side of an EC2 Transit Gateway VPC Attachment.
+
+When a cross-account (requester's AWS account differs from the accepter's AWS account) EC2 Transit Gateway VPC Attachment
+is created, an EC2 Transit Gateway VPC Attachment resource is automatically created in the accepter's account.
+The requester can use the `aws_ec2_transit_gateway_vpc_attachment` resource to manage its side of the connection
+and the accepter can use the `aws_ec2_transit_gateway_vpc_attachment_accepter` resource to "adopt" its side of the
+connection into management.
+
+## Example Usage
```
Comment (ewbankkit) [inline website/docs/r/ec2_transit_gateway_vpc_attachment_accepter.html.markdown]: Very simple example usage here; Full example add to `examples` folder.

**Comment by ewbankkit** [comment]
Removed _WIP_. Ready for review.

**Comment by ewbankkit** [comment]
I didn't add `transit_gateway_default_route_table_association` or `transit_gateway_default_route_table_propagation` attributes to this attribute because of the notes in the [documentation](https://www.terraform.io/docs/providers/aws/r/ec2_transit_gateway_vpc_attachment.html#transit_gateway_default_route_table_association):
> This cannot be configured or perform drift detection with Resource Access Manager shared EC2 Transit Gateways.

but maybe this resource, since it runs in the context of the AWS account that owns the transit gateway, is where these attributes should be specified in the cross-account case?

**Comment by ewbankkit** [comment]
Removed _WIP_.

Acceptance tests:

```console
$ AWS_ALTERNATE_ACCESS_KEY_ID=AAAAAAAAAAAAAAAAAAAA AWS_ALTERNATE_SECRET_ACCESS_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_ -timeout 120m
=== RUN   TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== PAUSE TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== RUN   TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation
=== PAUSE TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation
=== CONT  TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== CONT  TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation
--- PASS: TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic (366.19s)
--- PASS: TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation (555.40s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	555.436s
```

File: aws/resource_aws_ec2_transit_gateway_vpc_attachment_accepter.go [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+		if err := setTags(conn, d); err != nil {
+			return fmt.Errorf("error updating EC2 Transit Gateway VPC Attachment (%s) tags: %s", d.Id(), err)
+		}
+	}
+
+	return nil
+}
+
+func resourceAwsEc2TransitGatewayVpcAttachmentAccepterDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[WARN] Will not delete EC2 Transit Gateway VPC Attachment. Terraform will remove this resource from the state file, however resources may remain.")
```
Comment (bflad) [inline aws/resource_aws_ec2_transit_gateway_vpc_attachment_accepter.go]: The accepting side should be able to call `DeleteTransitGatewayVpcAttachment` here to break the attachment. 👍

File: aws/resource_aws_ec2_transit_gateway_vpc_attachment_accepter_test.go [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+					resource.TestCheckResourceAttrPair(resourceName, "transit_gateway_id", transitGatewayResourceName, "id"),
+					resource.TestCheckResourceAttrPair(resourceName, "transit_gateway_attachment_id", vpcAttachmentName, "id"),
+					resource.TestCheckResourceAttr(resourceName, "transit_gateway_default_route_table_association", "true"),
+					resource.TestCheckResourceAttr(resourceName, "transit_gateway_default_route_table_propagation", "true"),
+					resource.TestCheckResourceAttrPair(resourceName, "vpc_id", vpcResourceName, "id"),
+					resource.TestCheckResourceAttrPair(resourceName, "vpc_owner_id", callerIdentityDatasourceName, "account_id"),
+				),
+			},
+			{
+				Config: testAccAWSEc2TransitGatewayVpcAttachmentAccepterConfig_updated(rName),
```
Comment (bflad) [inline aws/resource_aws_ec2_transit_gateway_vpc_attachment_accepter_test.go]: The updated configuration here seems to be only modifying `tags`. We prefer the `_basic` tests to only check the required configuration for a resource (in this case, no tags) and creating a separate `Tags` test to check tags on create and update. 😄

File: website/docs/r/ec2_transit_gateway_vpc_attachment_accepter.html.markdown [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `transit_gateway_attachment_id` - (Required) The ID of the EC2 Transit Gateway Attachment to manage.
+* `transit_gateway_default_route_table_association` - (Optional) Boolean whether the VPC Attachment should be associated with the EC2 Transit Gateway association default route table. Default value: `true`.
+* `transit_gateway_default_route_table_propagation` - (Optional) Boolean whether the VPC Attachment should propagate routes with the EC2 Transit Gateway propagation default route table. Default value: `true`.
+* `tags` - (Optional) Key-value tags for the EC2 Transit Gateway VPC Attachment.
+
+### Removing `aws_ec2_transit_gateway_vpc_attachment_accepter` from your configuration
```
Comment (bflad) [inline website/docs/r/ec2_transit_gateway_vpc_attachment_accepter.html.markdown]: When the `Delete` function is implemented, this section of the resource documentation should be removed. If this restriction is actually necessary, please instead move it to a note at the top of the page:

```
~> **NOTE:** Deleting this resource only removes it from state management, etc.
```

**Comment by ewbankkit** [comment]
Review comments addressed.
Re-ran acceptance tests:

```console
$ AWS_ALTERNATE_ACCESS_KEY_ID=AAAAAAAAAAAAAAAAAAAA AWS_ALTERNATE_SECRET_ACCESS_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 2 -run=TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_ -timeout 120m
=== RUN   TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== PAUSE TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== RUN   TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_Tags
=== PAUSE TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_Tags
=== RUN   TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation
=== PAUSE TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation
=== CONT  TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic
=== CONT  TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation
--- PASS: TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_basic (332.64s)
=== CONT  TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_Tags
--- PASS: TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_TransitGatewayDefaultRouteTableAssociationAndPropagation (564.78s)
--- PASS: TestAccAWSEc2TransitGatewayVpcAttachmentAccepter_Tags (355.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	688.702s
```

**Comment by bflad** [comment]
This has been released in [version 2.13.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2130-may-31-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**